scripts/solve_network.py

```diff
@@ -358,21 +354,14 @@ def country_res_constraints(n, snakemake):
             target = float(snakemake.wildcards.res_share.replace("m","-").replace("p","."))
 
 
-        total_load = (n.loads_t.p_set[grid_loads].sum(axis=1)*weights).sum() # number
-
-        # add for ct in zone electrolysis demand to load if not "reference" scenario
-        # if (ct==zone) and (f"{ci_name}" in n.buses.index):
-
-        #     logger.info("Consider electrolysis demand for RES target.")
-        #     # H2 demand in zone
-        #     offtake_volume = float(snakemake.wildcards.offtake_volume)
-        #     # efficiency of electrolysis
-        #     efficiency = n.links[n.links.carrier=="H2 Electrolysis"].efficiency.mean()
-
-        #     # electricity demand of electrolysis
-        #     demand_electrolysis = (offtake_volume/efficiency
-        #                             *n.snapshot_weightings.generators).sum()
-        #     total_load += demand_electrolysis
+        el_load = (n.loads_t.p_set[grid_loads].sum(axis=1)*weights).sum() # number
+        
+        if ci_grid_loads.empty:
+            total_load = el_load
+        else:
+            h2_load = n.loads.p_set[ci_grid_loads].sum()*8760
+            h2_efficiency = n.links[n.links.carrier=="H2 Electrolysis"].efficiency.mean()
+            total_load = el_load + h2_load / h2_efficiency
```

Comment:
We can change the definition. That is how in theory the RES targets should be formulated. But I see two problems
1.) if you look at the German government plans on how they want to reach the 80% target they do not consider the electricity demand for 10 GW electrolysis what is set in the Fortsetzung deutscher Wasserstoffstrategie. Therefore, for example the set capacities targets in the background system don't fit to this higher electricity demand.
2.) from a modelling perspective, the installed capacities at the CI are not really additional. We will not see a difference between e.g. the grid scenario and the annual matching, since either the renewables are built in the background system or at the CI + background but the emissions will likely stay similar.